logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java

```diff
@@ -88,6 +100,74 @@ public void moveValidSnapshot(long lowerBoundSnapshotPosition) throws IOExceptio
     Files.move(previousLocation.toPath(), snapshotDir.toPath());
   }
 
+  public void replicateLatestSnapshot() {
+    final List<File> snapshots = storage.listByPositionDesc();
+
+    if (snapshots != null && !snapshots.isEmpty()) {
+      final File latestSnapshotDirectory = snapshots.get(0);
+      final long snapshotPosition = Long.parseLong(latestSnapshotDirectory.getName());
+
+      final File[] files = latestSnapshotDirectory.listFiles();
+      for (File snapshotChunk : files) {
```

Comment:
I assume that the snapshot replication can take some time. I could make sense to split the work and do not all at once. For example, replicate one file in one `actor.submit(...)`.